### PR TITLE
File upload fixes

### PR DESCRIPTION
**Pull request by PimCoumans**
Newly created transfers have their files ordered differently than requested, this was not accounted for, this is now happening for both transfers and boards. Instead of a `zip()` operation, local files and files from the response are paired one by one, and a custom error is thrown when there's a mismatch.
Also a incorrect endpoint was fixed for completing uploads

### CODE REVIEW

File: WeTransfer/Server/Endpoints/Endpoints.swift [hunk 1]
```diff
@@ -150,8 +150,10 @@ struct CreateTransferResponse: Decodable {
 		let id: String // swiftlint:disable:this identifier_name
 		/// Full name of file (e.g. "photo.jpg")
 		let name: String
+		// Size of the file in bytes
+		let size: Bytes
 		/// Mulitpart information about each chunk
-		let multipart: Multipart
+		let multipart: UploadInfo
```
Comment (AvdLee) [inline WeTransfer/Server/Endpoints/Endpoints.swift]: I would change the property name as well to multipartUploadInfo. Maybe call the struct `MultipartUploadInfo` as well to be more specific? @Boris-Em what do you think?

**Comment by AvdLee** [comment]
And thanks for the smaller PR! 🎉

File: WeTransfer/Server/Endpoints/Endpoints.swift (hunk 1 above)
Comment (PimCoumans) [reply WeTransfer/Server/Endpoints/Endpoints.swift]: `multipartUploadInfo` sounds like the best name for this. I’ll write it down for when I’ll set the `CodingKeys` for all requests and responses. For now the property will stay as the server expects it

Comment (PimCoumans) [reply WeTransfer/Server/Endpoints/Endpoints.swift]: `multipartUploadInfo` sounds like the best name for this. I’ll write it down for when I’ll set the `CodingKeys` for all requests and responses. For now the property will stay as the server expects it

Comment (PimCoumans) [reply WeTransfer/Server/Endpoints/Endpoints.swift]: `multipartUploadInfo` sounds like the best name for this. I’ll write it down for when I’ll set the `CodingKeys` for all requests and responses. For now the property will stay as the server expects it

Comment (PimCoumans) [reply WeTransfer/Server/Endpoints/Endpoints.swift]: `multipartUploadInfo` sounds like the best name for this. I’ll write it down for when I’ll set the `CodingKeys` for all requests and responses. For now the property will stay as the server expects it

**Pull request merged**